src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Change.cs

```diff
@@ -1,14 +1,170 @@
-using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
 
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
 
-public class Change<TState>
+public abstract class Change<TEntity>
 {
-    public Guid Identifier { get; set; } = UuidUtils.UuidV7();
+    public long Id { get; set; }
 
-    public required ChangeType Type { get; set; }
+    public DataSetVersion DataSetVersion { get; set; } = null!;
 
-    public TState? CurrentState { get; set; }
+    public required Guid DataSetVersionId { get; set; }
 
-    public TState? PreviousState { get; set; }
+    public TEntity? CurrentState { get; set; }
+
+    public TEntity? PreviousState { get; set; }
+
+    internal abstract class BaseConfig<TChange, TChangeEntity> : IEntityTypeConfiguration<TChange>
+        where TChange : Change<TChangeEntity>
+        where TChangeEntity : class
+    {
+        public virtual void Configure(EntityTypeBuilder<TChange> builder)
+        {
+            builder
+                .HasOne(c => c.CurrentState)
+                .WithMany()
+                .IsRequired(false);
+
+            builder
+                .HasOne(c => c.PreviousState)
+                .WithMany()
+                .IsRequired(false);
+
+            builder.Navigation(c => c.PreviousState).AutoInclude();
```

Comment:
Good idea - we've used this once before for DataBlocks but I do wonder sometimes if there are other places where we could leverage this.